### PR TITLE
fix: ajust widget displaying in conversation profile with advanced mode

### LINKED ISSUE #268: Apply the status of the advanced mode parameter in profile creation
When creating a profile, the advanced settings are displayed regardless of the status of the parameter in the application. I think this feature should be added. I can take care of it this weekend.

### PR DESCRIPTION
**Pull request by Nael-Sayegh**
closes #268 

## Summary by CodeRabbit

- **New Features**
	- Introduced advanced configuration options for conversation profiles, enhancing user control over settings.
	- Added methods to manage the visibility and adjustment of advanced options based on user preferences.

- **Bug Fixes**
	- Improved accessibility of UI element labels for better user experience.

- **Refactor**
	- Streamlined UI update process by consolidating methods related to advanced mode settings and improving organization of UI initialization code.

### CODE REVIEW

**Comment by Nael-Sayegh** [comment]
@coderabbitai How to create a single method for both the main window and the profile configuration dialog?

**Comment by Nael-Sayegh** [comment]
@aaclause @clementb49 How could we create a unique updateUI method for the different panels?

**Comment by aaclause** [comment]
@Nael-Sayegh Have you considered moving the `update_ui` method to the parent class `BaseConversation` so that the child classes, `ConversationTab` and `EditConversationProfileDialog`, can share this method? This would require normalizing the names of certain symbols in these child classes, such as `max_token_label` and `max_tokens_label`.
We should probably rename this method in this case to better reflect what it does. Perhaps something like `advanced_mode_aligner`?
Thanks!

File: basilisk/gui/conversation_tab.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def __init__(
 		self._search_dialog = None
 		self.init_ui()
 		self.init_data(profile)
-		self.update_ui()
+		self.display_advanced_mode()
```
[the remark this answers is not in the document]
Comment (Nael-Sayegh) [reply basilisk/gui/conversation_tab.py:93]: self.display_advanced_mode acts on UI elements in self.initUI, which is why `self.display_advanced_mode` is placed @coderabbiterai afterward. Do you think it is well-placed?

**Comment by Nael-Sayegh** [comment]
@aaclause What do you think?

**Comment by aaclause** [comment]
@Nael-Sayegh Quick test: the "Top P" field/label is broken, and is also labeled as "Max tokens".

**Comment by Nael-Sayegh** [comment]
@aaclause How can I manually restart the actions, as it seems there was an execution problem not related to the modifications?

**Comment by aaclause** [comment]
@Nael-Sayegh I just did it; no worries, everything is OK! We periodically have an issue with the cache that we need to clean manually...

File: basilisk/gui/base_conversation.py [hunk 2]
```diff
@@ -318,3 +318,19 @@ def apply_profile(
 		if profile.top_p is not None:
 			self.top_p_spinner.SetValue(profile.top_p)
 		self.stream_mode.SetValue(profile.stream_mode)
+
```
[the remark this answers is not in the document]
Comment (Nael-Sayegh) [reply basilisk/gui/base_conversation.py:321]: @coderabbitai display_advanced_mode only changes the user interface but does not modify the profile settings. I think this addition would be unnecessary.

**Comment by Nael-Sayegh** [comment]
@AAClause I think this PR is ready to be merged.

File: basilisk/gui/base_conversation.py [hunk 3]
```diff
@@ -318,3 +315,19 @@ def apply_profile(
...
+	def display_advanced_mode(self):
```
Comment (AAClause) [inline basilisk/gui/base_conversation.py:319]: To better reflect what the method does, I suggest the name `adjust_advanced_mode_setting`.

**Comment by AAClause** [comment]
Thanks @Nael-Sayegh!

**Pull request merged**